packages/backend/src/accounting/psql/service.ts

```diff
@@ -194,15 +216,279 @@ export async function getSettlementBalance(
+    sourceAccount,
+    sourceAssetAccount,
+    destinationAccount,
+    destinationAssetAccount,
+    sourceAmount,
+    destinationAmount,
+    timeout
+  } = args
+
+  if (!destinationAmount) {
+    return TransferError.InvalidDestinationAmount
+  }
+
+  const transfers: CreateTransferArgs[] = []
+
+  const addTransfer = ({
+    debitAccount,
+    creditAccount,
+    amount
+  }: {
+    debitAccount: LedgerAccount
+    creditAccount: LedgerAccount
+    amount: bigint
+  }) => {
+    transfers.push({
+      transferRef: uuid(),
+      debitAccount,
+      creditAccount,
+      amount,
+      timeoutMs: timeout
+    })
+  }
+
+  // Send to source liquidity account
+  addTransfer({
+    debitAccount: sourceAccount,
+    creditAccount: sourceAssetAccount,
+    amount: sourceAmount
+  })
+
+  // Deliver from destination liquidity account
+  addTransfer({
+    debitAccount: destinationAssetAccount,
+    creditAccount: destinationAccount,
+    amount: destinationAmount
+  })
+
+  return handleCreateTransferResults(
+    args,
+    transfers,
+    await createTransfers(deps, transfers)
+  )
+}
+
+async function handleCreateTransferResults(
+  args: CreateTransfersArgs,
+  attemptedTransfers: CreateTransferArgs[],
+  createTransferResults: CreateTransfersResult
+): Promise<LedgerTransfer[] | TransferError> {
+  const { errors, results: pendingTransfers } = createTransferResults
+
+  if (errors.length > 0) {
+    if (
+      errors.find(
+        ({ error, index }) =>
+          error === TransferError.InsufficientBalance &&
+          attemptedTransfers[index]?.debitAccount.id ===
+            args.destinationAssetAccount.id
+      )
+    ) {
+      return TransferError.InsufficientLiquidity
+    }
+
+    return errors[0].error
+  }
+
+  return pendingTransfers
+}
+
+function buildTransactionForTransfers(
+  deps: ServiceDependencies,
+  args: BuildTransactionForTransfers
+): Transaction {
+  const { destinationAccount, onDestinationAccountCredit, pendingTransfers } =
+    args
+
+  return {
+    post: async (): Promise<void | TransferError> => {
+      const results = await Promise.all(
+        pendingTransfers.map((transfer) =>
+          postTransfer(deps, transfer.transferRef)
+        )
+      )
+
+      const error = results.find((result) => isTransferError(result))
+
+      if (error) {
+        return error
+      }
+
+      if (onDestinationAccountCredit) {
+        const totalReceived = await getAccountTotalReceived(
+          deps,
+          destinationAccount.accountRef
+        )
+
+        if (totalReceived === undefined) {
+          throw new Error()
+        }
+
+        await onDestinationAccountCredit({
+          totalReceived,
+          withdrawalThrottleDelay: deps.withdrawalThrottleDelay
+        })
+      }
+    },
+    void: async (): Promise<void | TransferError> => {
+      const results = await Promise.all(
+        pendingTransfers.map((transfer) =>
+          voidTransfer(deps, transfer.transferRef)
+        )
```

Comment:
I think I could probably make a `voidTransferS` method to handle all of these in one go?

Reply:
Either that or we should pass a knex transaction to `voidTransfer`